Prevent extra space rendering under local pickup options in Checkout Block

opr opened the pull request:
### Changes proposed in this Pull Request:

- Add a new `selectedOption` arg for the `renderPickupLocation` method. We use to generate the react components that show the pickup instructions, however these instructions only show when the method is selected. If the method is not selected, the `ReadMore` component is still rendered, but it is empty, however there is spacing applied to it with CSS, even when empty.
- The method will skip rendering the component if it's not selected.

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->

(For Bug Fixes) Bug introduced in PR # .

### Screenshots or screen recordings:

<!-- If this PR includes UI changes, please provide screenshots or a screen recording for clarity. -->
<!-- This section can be removed if not applicable. -->

| Before | After |
| ------ | ----- |
| <img width="601" alt="image" src="https://github.com/user-attachments/assets/99fc8745-cddb-40c3-825f-3aea57bf6f27" /> |  <img width="609" alt="image" src="https://github.com/user-attachments/assets/f421d939-a529-41a3-b1c2-353363076358" /> |


<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to WC -> Settings -> Shipping -> Local Pickup and add multiple methods, some of these methods should have an empty "Pickup details" option.
<img width="565" alt="image" src="https://github.com/user-attachments/assets/724a947e-ccd6-4a80-b9da-1b6669802d5f" />

2. Add an item to your cart and go to the Checkout block.
3. Select Local Pickup and select different methods, ensure they show correctly.
4. Ensure other components on the page using Radio buttons work OK (Shipping, Payment).

<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

opr committed:
Add selectedOption value to renderPickupLocation

This is because we need to check the selected option to conditionally render the secondary description if it's selected

opr committed:
Pass selected option to renderPickupLocation

opr committed:
Skip rendering secondary description if local pickup method not selected

opr committed:
Add changelog

straku commented on plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/local-pickup-select/index.tsx:
```diff
@@ -47,7 +47,11 @@ export const LocalPickupSelect = ( {
 				selected={ selectedOption }
 				options={ pickupLocations.map( ( location ) =>
-					renderPickupLocation( location, packageCount )
+					renderPickupLocation(
```
Suggestion unrelated to the change – this should be a React component.

straku reviewed (approved):
Works for me, the fix looks straightforward. You might consider this an opportunity to increase test coverage and add a test, but I leave it up to you.

straku commented on plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/local-pickup-select/index.tsx:
```diff
@@ -47,7 +47,11 @@ export const LocalPickupSelect = ( {
+					renderPickupLocation(
+						location,
+						packageCount,
+						selectedOption
```
Day after feedback: instead of passing `selectedOption` here, it would be more fitting to pass `isSelected` instead. It's the parent component that decides which option is selected (hence having access to `selectedOption`), having all related logic to this aspect in the parent makes sense as well.

opr replied:
I think we can derive that from `location` so I removed the prop. Thanks!

opr committed:
Remove selectedOption from renderPickupLocation arg

opr committed:
Derive selected from the option itself, instead of passing the id prop

opr committed:
Remove unnecessary type

opr committed:
Add undefined as type.

The value *can* be undefined. The ? alone means the prop is optional, but if it is present it must be 'column' or 'row.

opr committed:
Add metadata to generateShippingRate mock

opr committed:
Ensure details don't show if rate is not selected
